src/new/wallet.js

```diff
@@ -89,28 +86,32 @@ class HathorWallet extends EventEmitter {
     this.passphrase = passphrase;
     this.pinCode = pinCode;
     this.password = password;
-  }
 
-  /**
-   * Called when loading the history of transactions.
-   * It is called every HTTP Request to get the history of a set of addresses.
-   * Usually, this is called multiple times.
-   * The `historyTransactions` contains all transactions, including the ones from a previous call to this method.
-   *
-   * @param {Object} result {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound}
-   **/
-  onAddressesLoaded(result) {
-    this.emit('more-addresses-loaded', result);
+    this.store = null;
+    if (store) {
+      this.store = store;
+    } else {
+      // Creating default store
+      this.store = new MemoryStore();
+    }
+    storage.setStore(this.store);
+
+    this.onConnectionChangedState = this.onConnectionChangedState.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
   }
 
   /**
    * Called when the connection to the websocket changes.
    * It is also called if the network is down.
+   *
+   * @param {Number} newState Enum of new state after change
    **/
-  onConnectionChange(value) {
-    if (value) {
+  onConnectionChangedState(newState) {
+    if (newState === Connection.CONNECTED) {
+      storage.setStore(this.store);
       this.setState(HathorWallet.SYNCING);
-      wallet.loadAddressHistory(0, GAP_LIMIT).then(() => {
+      WebSocketHandler.setup();
```

Comment:
I thought that if `newState === Connection.CONNECTED` the websocket was already setup. Wouldn't it be a problem to call this?

Reply:
It's a different websocket connection. When the refactor is done we will completely remove `WebSocketHandler` but for now it's needed to support the old library code.

In our normal sequence used nowadays in the wallet we call
`generateWallet` -> `WebSocketHandler.setup` -> `loadAddressHistory`

It's exactly what I did, except that I had to manually call the `WebSocketHandler.setup` because we don't automatically load the history. So it's not a problem, we are doing exactly what we do right now.